src/ducks/apps/components/Discover.jsx

```diff
@@ -49,6 +49,8 @@ export class Discover extends Component {
     const title = <h2 className="sto-view-title">{t('discover.title')}</h2>
     return (
       <Content className="sto-discover">
+        {isFetching && !RegExp('(.)*/discover/(.)+').test(window.location) && <AppsLoading />}
```

Comment:
Why about connecting this component to the router? And then use `location` from `props` it seems cleaner to me. To connecter the router, you can use
```js
import { withRouter } from 'react-router-dom'
...
export default translate()(withRouter(withBreakpoints()(Discover)))
```

and then you will be able to get `location` from `this.props` and do:
```suggestion
        {isFetching && !RegExp('(.)*/discover/(.)+').test(location) && <AppsLoading />}
```